desc/objectives/_generic.py

```diff
@@ -15,6 +19,262 @@
 from .objective_funs import _Objective
 
 
+class _ExternalObjective(_Objective, ABC):
+    """Wrap an external code.
+
+    Similar to ``ObjectiveFromUser``, except derivatives of the objective function are
+    computed with finite differences instead of AD.
```

Comment:
should also mention that the function need not be jax transformable/traceable etc.